torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -58,7 +58,8 @@ std::shared_ptr<JitFuture> RpcAgent::sendWithRetries(
       retryOptions.rpcRetryDuration.count() >= 0,
       "rpcRetryDuration cannot be negative.");
 
-  auto originalFuture = std::make_shared<JitFuture>(at::AnyClassType::get());
+  auto originalFuture =
+      std::make_shared<JitFuture>(at::AnyClassType::get(), getDevices());
```

Comment:
I assume we need `getDevices()` here so that `TensorPipeAgent` can override this API and enforce its own devices.

Reply:
Yes indeed here it made sense to me to create a future that had the same set of devices as what the agent supports, hence we needed a generic way to retrieve that set of devices. It also so happened that, for later in the stack, I also needed a way to retrieve the set of devices from the agent.

Another way we could have done this, by reordering this code, is to do `originalFuture = jitFuture->createInstance(at::AnyClassType::get());` I could go that way if you prefer.